x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/ClassificationInferenceResults.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ml.inference.results;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public class ClassificationInferenceResults extends SingleValueInferenceResults {
+
+    public static final String RESULT_TYPE = "classification";
+    public static final ParseField CLASSIFICATION_LABEL = new ParseField("classification_label");
```

Comment:
Is this the predicted class? If yes should we call it `predicted_class` instead?

Reply:
No, this field is just the optional string label for the ordinal value returned from the trained model.

Might be null as it is the specific string label for the numeric value returned via the model. If the user already transformed their classes into ordinal numerics, there is no label, just the numeric value.